### PR TITLE
Research page overhaul

### PR DESCRIPTION
**Pull request by Thodor12**
## Changes proposed:
- Adds README section on how to handle in-page links/names
- Fix 404 page
- Adds plural versions to both buildings and workers for used in display
- Adds all research into one big YAML file where everything can be easily configured
- Research pages is autogenerated, always looking 100% identical, properly including links to other pages, etc
- Individual researches now have ID anchors so they can be linked to from other pages (see links/research include)
- Removed the research tree images from the page as they had no real benefit
- Improved navigation on the page by adding "Back to the top" links in every research tree header

Merge please :D

### CODE REVIEW

**Comment by Thodor12** [comment]
125 looks like a scary big update but the majority is just include file renames

File: pages/src/_data/buildinginfo.yml [hunk 1]
```diff
@@ -31,6 +33,7 @@ bakery:
           description: The hut will look in the warehouse first to see what resource you have more of before deciding what recipe it will use.
 barracks:
   name: Barracks
```
Comment (anonymous563) [inline pages/src/_data/buildinginfo.yml:35]: Fun fact: the singular of "barracks" is "barrack."
But I almost never hear anyone refer to a single barrack.

File: pages/src/_data/workerinfo.yml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 alchemist:
   name: Alchemist
+  plural: Alchemists
   traits:
     primary: Dexterity?
```
Comment (anonymous563) [inline pages/src/_data/workerinfo.yml:5]: Can we confirm these? I had put question marks because I wrote this before they were out/publicly known.

Comment (Thodor12) [reply pages/src/_data/workerinfo.yml:5]: I can look at it tomorrow morning if those are actually what it says in the game aswell

Comment (anonymous563) [reply pages/src/_data/workerinfo.yml]: I just checked, it's right. I was hoping you had your game open already lol

**Review by anonymous563 — APPROVED** [review]

**Pull request merged**